packages/forma-36-codemod/transforms/v4-text-link.js

```diff
@@ -0,0 +1,84 @@
+const {
+  getComponentLocalName,
+  renameProperties,
+  hasProperty,
+  addProperty,
+  changeProperties,
+  updatePropertyValue,
+  updateIcons,
+  addIconImports,
+  changeImport,
+} = require('../utils');
+
+module.exports = function (file, api) {
+  const j = api.jscodeshift;
+
+  let source = file.source;
+
+  const componentName = getComponentLocalName(j, source, {
+    componentName: 'TextLink',
+    importName: '@contentful/forma-36-react-components',
+  });
+
+  const usedIcons = [];
+
+  source = changeProperties(j, source, {
+    componentName,
+    fn(attributes) {
+      let modifiedAttributes = attributes;
+
+      modifiedAttributes = renameProperties(modifiedAttributes, {
+        renameMap: {
+          linkType: 'variant',
+          disabled: 'isDisabled',
+          iconPosition: 'alignIcon',
+          text: 'children',
+        },
+      });
+
+      modifiedAttributes = updatePropertyValue(modifiedAttributes, {
+        j,
+        propertyName: 'alignIcon',
+        propertyValue: (value) => {
+          if (value.value === 'left') {
+            return j.literal('start');
+          } else if (value.value === 'right') {
+            return j.literal('end');
+          } else {
+            console.warn(
+              "TextLink: Couldn't automatically update alignIcon value",
+            );
+            return value;
+          }
+        },
+      });
+
+      if (hasProperty(modifiedAttributes, { propertyName: 'href' })) {
+        modifiedAttributes = addProperty(modifiedAttributes, {
+          j,
+          propertyName: 'as',
+          propertyValue: j.literal('a'),
+        });
+      }
+
+      if (hasProperty(modifiedAttributes, { propertyName: 'icon' })) {
+        modifiedAttributes = updateIcons(modifiedAttributes, {
+          j,
+          icons: usedIcons,
+        });
+      }
+
+      return modifiedAttributes;
+    },
+  });
+
+  source = changeImport(j, source, {
```

Comment:
Probably sometimes we don't want a change of the import to happen... For example, in Compose codebase. It looks like we need to make it controllable.